app/controllers/concerns/order_completion.rb

```diff
@@ -35,6 +35,10 @@ def order_completion_route
     main_app.order_path(@order, order_token: @order.token)
   end
 
+  def order_failed_route
+    main_app.checkout_path
```

Comment:
No need to pass the `@order.state` here?

Reply:
Nope. That reminds me; I think there's still various places which use the alternative, which is: `checkout_state_path(@order.state)`. I think we can actually just delete that route definition completely. All it does is go to the `checkout_path` (the main edit checkout page) but also passes `params[:state]` to the controller... but that param isn't actually used anywhere in the controller... :wastebasket:

Reply:
IO think it may have been used by Spree a long time ago, but we don't use it :man_shrugging: